tests/test_arrow_dataset.py

```diff
@@ -2708,6 +2708,154 @@ def test_tf_dataset_conversion(self, in_memory):
+                tf_dataset = dset.to_tf_dataset(
+                    columns="col_1", label_cols="col_2", batch_size=2, num_workers=num_workers
+                )
+                batch = next(iter(tf_dataset))
+                self.assertTrue(len(batch) == 2)
+                self.assertEqual(batch[0].shape.as_list(), [2])
+                self.assertEqual(batch[0].dtype.name, "int64")
+                self.assertEqual(batch[1].shape.as_list(), [2])
+                self.assertEqual(batch[1].dtype.name, "string")
+                # If return_dict is None, and there's more than one feature column, we get a tuple, where the first element is a dict
+                # and the second element is a tuple
+                tf_dataset = dset.to_tf_dataset(
+                    columns=["col_1", "col_3"], label_cols="col_2", batch_size=2, num_workers=num_workers
+                )
+                batch = next(iter(tf_dataset))
+                self.assertEqual(batch[0]["col_1"].shape.as_list(), [2])
+                self.assertEqual(batch[0]["col_1"].dtype.name, "int64")
+                self.assertEqual(batch[0]["col_3"].shape.as_list(), [2])
+                self.assertEqual(batch[0]["col_3"].dtype.name, "int64")
+                self.assertEqual(batch[1].shape.as_list(), [2])
+                self.assertEqual(batch[1].dtype.name, "string")
+                # If return_dict is None, and there's more than one label column, we get tuple, where the first element is a tuple
+                # and the second element is a dict
+                tf_dataset = dset.to_tf_dataset(
+                    columns="col_1", label_cols=["col_2", "col_3"], batch_size=2, num_workers=num_workers
+                )
+                batch = next(iter(tf_dataset))
+                self.assertEqual(batch[0].shape.as_list(), [2])
+                self.assertEqual(batch[0].dtype.name, "int64")
+                self.assertEqual(batch[1]["col_2"].shape.as_list(), [2])
+                self.assertEqual(batch[1]["col_2"].dtype.name, "string")
+                self.assertEqual(batch[1]["col_3"].shape.as_list(), [2])
+                self.assertEqual(batch[1]["col_3"].dtype.name, "int64")
+                # If return_dict is True we get a dict for all combinations of feature and label columns
+                tf_dataset = dset.to_tf_dataset(
+                    columns="col_1", return_dict=True, batch_size=2, num_workers=num_workers
+                )
+                batch = next(iter(tf_dataset))
+                self.assertIsInstance(batch, dict)
+                self.assertEqual(batch["col_1"].shape.as_list(), [2])
+                self.assertEqual(batch["col_1"].dtype.name, "int64")
+
+                tf_dataset = dset.to_tf_dataset(
+                    columns="col_1", label_cols="col_2", return_dict=True, batch_size=2, num_workers=num_workers
+                )
```

Comment:
I find it confusing to set `return_dict=True` and return a tuple.

In that case the user should to `columns=["col_1", "col_2"], return_dict=True` no ?

IMO you can raise an error if users pass both `label_cols=` and `return_dict=True`